corehq/apps/api/resources/v0_5.py

```diff
@@ -277,15 +278,33 @@ def obj_create(self, bundle, **kwargs):
             validate_profile_required(bundle.data.get('user_profile'), kwargs['domain'])
         except ValidationError as e:
             raise BadRequest(e.message)
+        require_account_confirmation = True if bundle.data.get('require_account_confirmation') == "True" else False
+        send_confirmation_email = True if bundle.data.get('send_confirmation_email_now') == "True" else False
         try:
-            bundle.obj = CommCareUser.create(
-                domain=kwargs['domain'],
-                username=username,
-                password=bundle.data.get('password'),
-                created_by=bundle.request.couch_user,
-                created_via=USER_CHANGE_VIA_API,
-                email=bundle.data.get('email', '').lower(),
-            )
+            email = bundle.data.get('email', '').lower()
+            if toggles.TWO_STAGE_USER_PROVISIONING.enabled(kwargs['domain']):
+                if require_account_confirmation and send_confirmation_email and not email:
+                    raise BadRequest(_("You must provide the user's email to send a confirmation email."))
+                bundle.obj = CommCareUser.create(
+                    domain=kwargs['domain'],
+                    username=username,
+                    password=bundle.data.get('password'),
```

Comment:
This doesn't seem right. The password shouldn't be defined by the user if `require_account_confirmation` is true. But i'm not sure what it should be set to. I don't think `None` is correct. Could check how it's being handled in the manual workflow